src/portfft/committed_descriptor_impl.hpp

```diff
@@ -242,17 +224,29 @@ class committed_descriptor_impl {
     std::shared_ptr<IdxGlobal> factors_and_scan;
     detail::level level;
     std::size_t length;
+    std::size_t committed_length;
```

Comment:
Committed length vs length is going to cause confusion. Either improve the name or describe them better.